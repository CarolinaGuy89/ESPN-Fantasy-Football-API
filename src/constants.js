/**
 * Maps `slotCategoryId`'s numerical enum to readable positions.
 * @type {Object}
 */
const slotCategoryIdToPositionMap = {
  0: 'QB',
  1: 'TQB',
  2: 'RB',
  3: 'RB/WR',
  4: 'WR',
  5: 'WR/TE',
  6: 'TE',
  7: 'OP',
  8: 'DT',
  9: 'DE',
  10: 'LB',
  11: 'DL',
  12: 'CB',
  13: 'S',
  14: 'DB',
  15: 'DP',
  16: 'D/ST',
  17: 'K',
  18: 'P',
  19: 'HC',
  20: 'Bench',
  21: 'IR',
  22: 'Unknown?', // TODO: Figure out what this is
  23: 'RB/WR/TE',
  24: 'Unknown?' // TODO: Figure out what this is
};

export { slotCategoryIdToPositionMap };
