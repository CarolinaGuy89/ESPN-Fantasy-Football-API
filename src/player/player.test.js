import _ from 'lodash';

import BaseObject from '../base-object/base-object.js';

import Player from './player.js';

import { localObject, serverResponse } from './player.stubs.js';

import { slotCategoryIdToPositionMap } from '../constants.js';

describe('Player', () => {
  test('extends BaseObject', () => {
    const instance = new Player();
    expect(instance).toBeInstanceOf(BaseObject);
  });

  describe('when creating a team from a server response', () => {
    test('parses and assigns data correctly', () => {
      const instance = Player.buildFromServer(serverResponse);
      expect(instance).toMatchSnapshot();
    });
  });

  describe('when creating a team locally', () => {
    test('parses and assigns data correctly', () => {
      const instance = new Player(localObject);
      expect(instance).toMatchSnapshot();
    });
  });

  describe('responseMap', () => {
    const buildPlayer = (data, options) => Player.buildFromServer(data, options);

    describe('streakType', () => {
      describe('manualParse', () => {
        test('maps ids to positions', () => {
          const eligibleSlotCategoryIds = [0, 1, 2];
          const data = { eligibleSlotCategoryIds };

          const player = buildPlayer(data);

          expect.hasAssertions();
          _.forEach(player.eligiblePositions, (position, index) => {
            expect(position).toBe(
              _.get(slotCategoryIdToPositionMap, eligibleSlotCategoryIds[index])
            );
          });
        });
      });
    });
  });

  describe('class methods', () => {
    describe('getCacheId', () => {
      test('returns undefined', () => {
        expect(Player.getCacheId()).toBeUndefined();
      });
    });
  });
});
