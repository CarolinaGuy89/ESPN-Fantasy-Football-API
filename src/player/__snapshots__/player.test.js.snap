// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player when creating a team from a server response parses and assigns data correctly 1`] = `
Player {
  "eligiblePositions": Array [
    "QB",
    "Bench",
  ],
  "firstName": "Aaron",
  "isActive": true,
  "isIREligible": false,
  "jerseyNumber": "12",
  "lastName": "Rodgers",
  "percentOwned": 99.11713,
  "percentOwnedChange": 0.16773,
  "percentStarted": 68.05881,
  "playerId": 8439,
}
`;

exports[`Player when creating a team locally parses and assigns data correctly 1`] = `
Player {
  "eligiblePositions": Array [
    "QB",
    "Bench",
  ],
  "firstName": "Aaron",
  "isActive": true,
  "isIREligible": false,
  "jerseyNumber": "12",
  "lastName": "Rodgers",
  "percentOwned": 99.11713,
  "percentOwnedChange": 0.16773,
  "percentStarted": 68.05881,
  "playerId": 8439,
}
`;
